### ISSUE TITLE
`Union[X, NoReturn]` isn't properly narrowed to `X`

### ISSUE THREAD

**Issue by tusharsadhwani** [python/mypy#11876]
```python
from typing import Union, NoReturn

def func() -> Union[str, NoReturn]:
    return 'test'

func().lower()
```

Here, `Unoin[str, NoReturn]` should effectively be treated as the same as `str`. But mypy doesn't seem to do that.

```console
$ mypy code.py
code.py:6: error: Item "NoReturn" of "Union[str, NoReturn]" has no attribute "lower"
Found 1 error in 1 file (checked 1 source file)
```

More information in this discussion page: https://github.com/python/typing/discussions/994#discussioncomment-1871267

**Comment by gschaffner** [comment]
This appears to be fixed as of Mypy 0.940.

**Comment by hauntsaninja** [comment]
Thanks, this was fixed in https://github.com/python/mypy/pull/11996
